skbeam/core/recip.py

```diff
@@ -230,3 +231,113 @@ def calibrated_pixels_to_q(detector_size, pyfai_kwargs):
                            "function.")
     a = geo.Geometry(**pyfai_kwargs)
     return a.qArray(detector_size)
+
+
+gisaxs_output = namedtuple(
+    'gisaxs_output',
+    ['alpha_i', 'theta_f',
+     'alpha_f', 'tilt_angle',
+     'qx', 'qy', 'qz', 'qr']
+)
+
+
+def gisaxs(incident_beam, reflected_beam, pixel_size, detector_size,
+           dist_sample, wavelength, theta_i=0.0):
+    """
+    This function will provide scattering wave vector(q) components(x, y, z),
+    q parallel and incident and reflected angles for grazing-incidence small
+    angle X-ray scattering (GISAXS) geometry.
+
+    Parameters
+    ----------
+    incident_beam : tuple
+        x and y co-ordinates of the incident beam in pixels
+    reflected_beam : tuple
+        x and y co-ordinates of the reflected beam in pixels
+    pixel_size : tuple
+        pixel_size in um
+    detector_size: tuple
+        2 element tuple defining no. of pixels(size) in the
+        detector X and Y direction
+    dist_sample : float
+       sample to detector distance, in meters
+    wavelength : float
+        wavelength of the x-ray beam in Angstroms
+    theta_i : float, optional
+        out of plane angle, default 0.0
+
+    Returns
+    -------
+    alpha_i : float
```

Comment:
Can you note that this actually returns a named tuple whose contents are this list of return values